### PR TITLE
 fix(build): bump `@octokit/tsconfig` major version

### PR DESCRIPTION
**Pull request by kfcampbell**
This PR uses [auth-token.js#345](https://github.com/octokit/auth-token.js/pull/345) as a model to update the major octokit/tsconfig version, revert the breaking engines change (that will take place during the upcoming Node 14/16 deprecation), and adds a test-specific tsconfig.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -46,6 +46,14 @@
   },
   "jest": {
     "preset": "ts-jest",
```
Comment (wolfy1339) [inline package.json:48]:
```suggestion
```

**Comment by wolfy1339** [comment]
The PR for Node 14/16 removal was merged here already

File: package.json [hunk 2]
```diff
@@ -88,6 +95,6 @@
     ]
   },
   "engines": {
-    "node": ">= 18"
+    "node": ">= 14"
```
Comment (kfcampbell) [inline package.json:98]:
```suggestion
    "node": ">= 18"
```

**Comment by kfcampbell** [comment]
Ahh oops, thanks!

**Review by wolfy1339 — APPROVED** [review]

**Pull request merged**